SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/TypeStructure.cpp

```diff
@@ -259,6 +259,42 @@ TSharedPtr<FUnrealType> CreateUnrealTypeInfo(UStruct* Type, uint32 ParentChecksu
 		return TypeNode;
 	}
 
+	if (Class->IsChildOf<AActor>())
+	{
+		const AActor* CDO = Cast<AActor>(Class->GetDefaultObject());
+
+		// Handle components attached to the actor that don't have a property pointing to them.
+		for (UActorComponent* Component : CDO->GetComponents())
+		{
+			if (Component->IsEditorOnly())
+			{
+				continue;
+			}
+
+			if (!Component->IsSupportedForNetworking())
+			{
+				continue;
+			}
+
+			TSharedPtr<FUnrealProperty> PropertyNode = MakeShared<FUnrealProperty>();
+			PropertyNode->Property = nullptr;
```

Comment:
Yuck - breaks the contract of Property never being `nullptr`

Reply:
Does this need to be an `FUnrealProperty`? It looks like we only use `Type` later on, so perhaps we can just store the `FUnrealType`s instead?

Reply:
I'm experimenting with something similar in a temp branch, does seem like we can do just that.

Reply:
Done by adding a new struct `FUnrealSubobject`.